### PR TITLE
refactor: update to Node 20

### PR DESCRIPTION
**Pull request by mfranzke**
## Proposed changes

Resolves https://github.com/db-ui/mono/issues/1823

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by mfranzke** [comment]
The problem coming up is related to the comment `node --experimental-specifier-resolution=node scripts/generate-docs-mdx.js` being run within the `showcases/patternhub` folder.

File: showcases/patternhub/scripts/generate-docs-mdx.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-/* eslint-disable unicorn/prefer-top-level-await, no-await-in-loop */
+/* eslint-disable unicorn/prefer-top-level-await, unicorn/prefer-node-protocol, no-await-in-loop */
 
-import FS from 'node:fs';
+import FS from 'fs';
```
Comment (mfranzke) [inline showcases/patternhub/scripts/generate-docs-mdx.js:3]: what's the reason for these changes?

Comment (nmerget) [reply showcases/patternhub/scripts/generate-docs-mdx.js]: If I use `esbuild` wiht `--platform=node` I get some warnings about `hooks` and a missmatch between React version. If I use `--platform=browser` it works as expected.
Don't ask me why...
But `xo --fix` would add `node:fs` which would break a "browser" build

Comment (mfranzke) [reply showcases/patternhub/scripts/generate-docs-mdx.js]: > If I use `esbuild` wiht `--platform=node` I get some warnings about `hooks` and a missmatch between React version. If I use `--platform=browser` it works as expected. Don't ask me why... But `xo --fix` would add `node:fs` which would break a "browser" build

Seems to be a Windows topic …

Comment (mfranzke) [reply showcases/patternhub/scripts/generate-docs-mdx.js]: Solved the remaining problem reported by the pipeline through this solution:
https://github.com/evanw/esbuild/issues/1921#issuecomment-1152991694

**Review by nmerget — APPROVED** [review]
